public/app/features/library-panels/components/LibraryPanelsSearch/LibraryPanelsSearch.tsx

```diff
@@ -18,43 +28,62 @@ export interface LibraryPanelsSearchProps {
   onClick: (panel: LibraryPanelDTO) => void;
   variant?: LibraryPanelsSearchVariant;
   showSort?: boolean;
-  showFilter?: boolean;
+  showPanelFilter?: boolean;
+  showFolderFilter?: boolean;
   showSecondaryActions?: boolean;
   currentPanelId?: string;
+  currentFolderId?: number;
   perPage?: number;
 }
 
 export const LibraryPanelsSearch = ({
   onClick,
   variant = LibraryPanelsSearchVariant.Spacious,
   currentPanelId,
+  currentFolderId,
   perPage = DEFAULT_PER_PAGE_PAGINATION,
-  showFilter = false,
+  showPanelFilter = false,
+  showFolderFilter = false,
   showSort = false,
   showSecondaryActions = false,
 }: LibraryPanelsSearchProps): JSX.Element => {
-  const [searchQuery, setSearchQuery] = useState('');
-  const [sortDirection, setSortDirection] = useState<string | undefined>(undefined);
-  const [panelFilter, setPanelFilter] = useState<string[]>([]);
   const styles = useStyles2(getStyles);
-  const onSortChange = useCallback((sort: SelectableValue<string>) => setSortDirection(sort.value), []);
-  const onFilterChange = useCallback((plugins: PanelPluginMeta[]) => setPanelFilter(plugins.map((p) => p.id)), []);
+  const [{ sortDirection, panelFilter, folderFilter, searchQuery }, dispatch] = useReducer(libraryPanelsSearchReducer, {
+    ...initialLibraryPanelsSearchState,
+    folderFilter: currentFolderId ? [currentFolderId.toString(10)] : [],
+  });
+  const onFilterChange = (searchString: string) => dispatch(searchChanged(searchString));
+  const onSortChange = (sorting: SelectableValue<string>) => dispatch(sortChanged(sorting));
+  const onFolderFilterChange = (folders: FolderInfo[]) => dispatch(folderFilterChanged(folders));
+  const onPanelFilterChange = (plugins: PanelPluginMeta[]) => dispatch(panelFilterChanged(plugins));
```

Comment:
Interesting how you married React reducers with @redux/toolkit created actions, this is very neat. Sounds like a topic for a Frontend weekly to me, WDYT?

Reply:
Yepp, I could also use `createSlice` here now when I think about it. Maybe a Frontend weekly after v8 😉

Reply:
Yepp, I could also use `createSlice` here now when I think about it. Maybe a Frontend weekly after v8 😉

Reply:
of course, no pressure!